### ISSUE TITLE
Vault integration not working properly? Towny/Essential issues

### ISSUE THREAD

**Issue by Blackyvk** [pavog/craftconomy3#33]
After testing your pugin with these two, I dont know if other plugins missbehave.

**Towny**
Creating a towny town, doesnt works, 
First it adds uuids to town accounts, which it shouldnt as towns arent players
Second it generates this error:
https://pastebin.com/raw/YbHPvQ1C

**EssentialsX,**

Balances are not sync, ie:
/eco give doesnt add money to the craftconomy accounts. 

All this worked on older versions of the plugin even on 1.13.2.

spigot 1.13.2
vault 1.7.1
craftconomy 3.5.1

**Comment by pavog** [comment]
**Towny**

I found out that Towny uses balances for towns. So it handles towns like players and tries to update their balance like a player. Unfortunately craftconomy tries to resolve the world of the player to check which balance / account it should use and a town does not have a location and world like a player has.
I need to implement a workaround for this location / world check if the checked object is not a player or just use a more general check for the location.


EssentialsX

I think that EssentialsX uses their own economy system. Since /eco is a command that belongs to EssentialsX and EssentialsX does not sync with any other economy plugin because it uses their own economy plugin this behaviour is the expected.

**Comment by Blackyvk** [comment]
Yes essentials has a economy system, but when other economy plugin is added, they both are usually in sync, on CraftConomy 3.3.1 worked like that, and I think it's because essentials economy cannot be disabled and to avoid issues with other plugins they both work in sync? maybe.

**Comment by pavog** [comment]
Ah yes, I found [this article](http://wiki.mc-ess.net/wiki/Essentials_Economy/API) that proves your assumption. It seems like the connection from Vault to CraftConomy does not work like expected. I need to investigate further on this.

**Comment by Weark** [comment]
I'm having similar problems, and I do not want to abandon the craftcononey3 plugin because it's great.
Once you fix, compile, and release the download for everyone, it will help a lot of people.
Thank you.
